sips/sip-026/sip-026-clarity-defi-vault.md

```diff
@@ -0,0 +1,97 @@
+# Preamble
+
+SIP-Number: SIP-026
+Title: Clarity DeFi Vault
+Author: Tycho Onnasch, Fernando Foy, Philip de Smedt, Christian Hresko
+Consideration: Technical
+Type: Standard
+Status: Accepted
+Created: Sep 7, 2023
+Last-Modified: Sep 7, 2023
+Sign-off:
+Discussions-To: https://forum.stacks.org/t/clarity-defi-vault-sip/15567
+License: Creative Commons CC0 1.0 Universal license
+Layer: Trait
+
+# Abstract
+
+This Stacks Improvement Proposal (SIP) aims to address the issue of commingled collateral positions on the Stacks blockchain's Clarity-based DeFi applications. Currently, due to Clarity's design, it is challenging for users to identify their specific collateral positions within the Stacks Explorer. This SIP proposes the creation of a common interface for contracts that hold SIP-010 assets. Implementing this interface will enable users to view their collateral positions distinctly, improving the overall user experience of Stacks DeFi applications.
+
+# License and Copyright
+
+This SIP is made available under the terms of the Creative Commons CC0 1.0 Universal license, available at https://creativecommons.org/publicdomain/zero/1.0/. This SIP's copyright is held by the Stacks Open Internet Foundation.
+
+# Introduction
+
+#### Problem Statement
+Clarity, the smart contract language of the Stacks blockchain, presents a challenge in providing users with the ability to identify their collateral positions within Clarity-based DeFi applications. Unlike other blockchain ecosystems like Ethereum, where users can distinctly view their collateral in separate smart contracts on the chain's explorer, Clarity currently lacks this capability.
+
+As a result, users who post collateral on Stacks DeFi apps might perceive their collateral funds as commingled with other users' collateral, leading to a suboptimal user experience. The ability to natively inspect collateral positions on-chain is a crucial value proposition of DeFi over CeFi.
+
+#### Technical Background of the Problem
+In Solidity, the Ethereum smart contract language, contracts can be deployed by executing Solidity code in transactions. This allows the deployment of contracts that represent vaults holding ERC-20s separately. These distinct collateral-holding contracts can be effortlessly looked up on explorers such as Etherscan, facilitating users' oversight of DeFi protocol collateralization.
+
+However, Clarity's design differs significantly. It does not allow contract deployment during transaction execution to maintain deterministic behavior and avoid infinite recursion. Consequently, representing entities that hold SIP-010 assets requires implementing logic within a single contract address.
+
+# Specification
```

Comment:
I think it would be good to talk about event production (via `print`) for vaults.

I imagine that a standardized deposit/withdraw event for vaults could eventually be used by explorers to display a more detailed event outcome for a transaction. For example, rather than `transfered FOO to SP1VFTXPFH4S8T99NHWQBJ8BECKZFQ3YFXRN4NJY8.bns` it could say `transfered FOO to SP1VFTXPFH4S8T99NHWQBJ8BECKZFQ3YFXRN4NJY8.bns, depositing in vault 10` or whatever. But it can only do that if the vault events are standardized.